pkg/streamingpromql/testdata/ours/functions.test

```diff
@@ -940,3 +940,30 @@ eval range from 0 to 4m step 1m sort_desc(test_metric)
   test_metric{case="histogram with NaN"}  {{count:0 sum:NaN}} {{count:0 sum:NaN}} {{count:0 sum:NaN}} {{count:0 sum:NaN}} {{count:0 sum:NaN}}
   test_metric{case="histogram with +Inf"} {{count:0 sum:Inf}} {{count:0 sum:Inf}} {{count:0 sum:Inf}} {{count:0 sum:Inf}} {{count:0 sum:Inf}}
   test_metric{case="histogram with -Inf"} {{count:0 sum:-Inf}} {{count:0 sum:-Inf}} {{count:0 sum:-Inf}} {{count:0 sum:-Inf}} {{count:0 sum:-Inf}}
+
+clear
+
+# Test edge cases for quantile_over_time.
+load 1m
+  metric{case="floats"} 1 2
+  metric{case="NaN at end"} 1 NaN
+  metric{case="NaN at start"} NaN 2
+  metric{case="all NaN"} NaN NaN
+  metric{case="Inf at start"} 1 Inf
+  metric{case="Inf at end"} Inf 2
+  metric{case="all Inf"} Inf Inf
+  metric{case="-Inf at start"} 1 -Inf
+  metric{case="-Inf at end"} -Inf 2
+  metric{case="all -Inf"} -Inf -Inf
+
+eval instant at 1m quantile_over_time(0.5, metric[1m1s])
```

Comment:
some tests where the scalar changes over time could be interesting. The gauntlet is probably sufficient though if done there

Reply:
Added in b0fceec.